sympy/combinatorics/fp_groups.py

```diff
@@ -407,6 +479,33 @@ def process_deductions(self, R_c_x, R_c_x_inv):
                     if p[beta] < beta:
                         break
 
+    # Pg. 166
+    def process_deductions_check(self, R_c_x, R_c_x_inv):
+        """
+        A variation of "process_deductions", this calls "scan_check" wherever
+        "process_deductions" calls "scan".
+        """
+        p = self.p
+        while len(self.deduction_stack) > 0:
+            if len(self.deduction_stack) >= max_stack_size:
+                self.lookahead()
+                del self.deduction_stack[:]
+            else:
+                alpha, x = self.deduction_stack.pop()
+                if p[alpha] == alpha:
+                    for w in R_c_x:
+                        self.scan_check(alpha, w)
```

Comment:
We should probably have something like `if not self.scan_check(alpha, w): return False`.
It also seems that the next line will not be needed since `scan_check` does not modify `p`.